packages/dso-toolkit/gulp/build-svg-spritesheets.js

```diff
@@ -0,0 +1,165 @@
+                  })
+                  .filter(Boolean); // Remove null entries
+              });
+
+            return { id, variants };
+          });
+
+          resolve(parsedStyles);
+        }),
+      );
+  });
+}
+
+async function buildDiSpritesheet() {
+  const iconStylesheets = await parseStylesheets(); // Parse stylesheets within the function
+
+  if (!iconStylesheets || iconStylesheets.length === 0) {
+    throw new Error("No icon stylesheets found. Pipeline terminated.");
+  }
+
+  return gulp
+    .src(`${iconsPath}/*.svg`)
+    .pipe(
+      svgstore({
+        inlineSvg: true,
+      }),
+    )
+    .pipe(
+      cheerio({
+        run($) {
+          const symbols = $("symbol").toArray();
+          let position = 0;
+
+          // Create a map for quick stylesheet lookup
+          const stylesheetMap = Object.fromEntries(iconStylesheets.map((s) => [s.id, s]));
+
+          for (const symbol of symbols) {
+            const $symbol = $(symbol);
+            const id = $symbol.attr("id");
+            const stylesheet = stylesheetMap[id];
+
+            if (!stylesheet) {
+              $symbol.remove();
+              continue;
+            }
+
+            const variants = [
+              { selector: id, color: stylesheet.variants.find((v) => v.selector === "default")?.color },
+              ...stylesheet.variants
+                .filter((v) => v.selector !== "default")
+                .map((v) => ({ selector: `${id}-${v.selector}`, color: v.color })),
+            ];
+
+            for (const [index, variant] of variants.entries()) {
+              const x = (position + index) * (canvasSize + gutterSize) + gutterSize / 2;
+
+              const view = $("<view>")
+                .attr("id", `${variant.selector}`)
+                .attr("viewBox", [x, 0, canvasSize, canvasSize].join(" "));
+
+              const svgElement = $("<svg>").attr("id", variant.selector).attr("viewBox", $symbol.attr("viewBox"));
+
+              // Convert children to an array and clone each child
+              for (const child of $symbol.children().toArray()) {
+                const clonedChild = $(child).clone();
+                if (variant.color && clonedChild.attr("fill") === "currentColor") {
+                  clonedChild.attr("fill", variant.color);
+                }
+                svgElement.append(clonedChild);
+              }
+
+              const group = $("<g>").attr("transform", `translate(${x})`).append(svgElement);
+
+              $symbol.before(view).before(group);
+            }
+
+            $symbol.remove();
+            position += variants.length;
+          }
+
+          // Calculate and set the parent SVG viewBox
+          const totalWidth = position * (canvasSize + gutterSize) + gutterSize;
+          $(":root").attr("viewBox", `0 0 ${totalWidth} ${canvasSize}`);
+        },
+        parserOptions: { xmlMode: true },
+      }),
+    )
+    .pipe(prettier({ parser: "html" }))
```

Comment:
Waar komt parser: html ineens vandaan? Hier hebben we speciaal de de XML Prettier plugin voor geinstalleerd:

```
.pipe(prettier({ plugins: ["@prettier/plugin-xml"] }))
```

Reply:
Ah, ik begrijp het denk ik: De xml printer maakt er nogal een lange brei van